deploy/kubernetes/helm/che/templates/deployment.yaml

```diff
@@ -100,24 +100,24 @@ spec:
           valueFrom:
             secretKeyRef:
               key: ca.crt
-              name: {{ .Values.global.tls.selfSignedCertSecretName }}
+              name: {{ .Values.global.tls.secretName }}
               optional: false
         {{- end }}
 
         # If workspaces are created in a separate precreated namespace
         # then configure Che Server to propagate TLS secret to workspaces' namespaces
-        {{- if not (contains "<" .Values.global.cheWorkspacesNamespace) }}
+        {{- if contains "<" .Values.global.cheWorkspacesNamespace }}
```

Comment:
So originally, before we had placeholders in namespace names, this test was merely "if the namespace is different than the default one". With the introduction of placeholders, we changed the condition to "if this is a static, precreated, namespace". Reverting that condition as you suggest, would actually mean "if this is either a precreated namespace using a placeholder (e.g. `<username>-che`) or a namespace we create (e.g. `<workspaceid>-che`)".

So I don't think this change is actually valid. At the same time I agree the test is insufficient :) We need to capture the "precreatedness" of the namespaces here and I am afraid we cannot capture that by merely detecting the placeholders in the name.

Maybe just actually give up and introduce `global.namespacePrecreated`?

Reply:
I wrote a lot of words and thought about the behavior we need here, but then just removed everything because I got to a conclusion that Che Server must be configured with TLS certs to propagate always if it's different than Che Server one...

We introduce placeholders and some conditions in helm chart and they are valid for ServiceAccounts, Roles, and RoleBindings... But what is the reason not to propagate TLS *secret* for WorkspaceIngresses?... At this stage, I believe that they must be propagated always if workspaces are not deployed in the same namespace as Che is because only in that case workspaces are not able to reuse Che Server's one... 
Hope I described my thought quite clearly. Let me fix condition as I described and then we'll continue discussing it